paddle/fluid/operators/mkldnn/interpolate_mkldnn_op.cc

```diff
@@ -0,0 +1,208 @@
+    } else if (in_dhw_dims.size() == 3) {
+      out_dims.push_back(ctx.Attr<int>("out_d"));
+      out_dims.push_back(ctx.Attr<int>("out_h"));
+      out_dims.push_back(ctx.Attr<int>("out_w"));
+    }
+    PADDLE_ENFORCE_GT(
+        std::all_of(out_dims.begin(), out_dims.end(),
+                    [](int i) { return i > 0; }),
+        0, platform::errors::InvalidArgument(
+               "out_d, out_h, out_w in Attr(out_shape) of Op(interpolate) "
+               "should be greater than 0."));
+
+    auto list_new_size_tensor = ctx.MultiInput<framework::Tensor>("SizeTensor");
+    if (list_new_size_tensor.size() > 0) {
+      auto new_size = get_new_shape(list_new_size_tensor);
+      if (new_size.size() == out_dims.size()) {
+        out_dims = new_size;
+      }
+    } else {
+      float scale;
+      auto scale_tensor = ctx.Input<Tensor>("Scale");
+      if (scale_tensor != nullptr) {
+        auto scale_data = get_new_data_from_tensor<float>(scale_tensor);
+        scale = scale_data[0];
+      } else {
+        scale = ctx.Attr<float>("scale");
+      }
+      if (scale > 0) {
+        std::vector<int64_t> in_dhw_vec = framework::vectorize(in_dhw_dims);
+        std::transform(
+            in_dhw_vec.begin(), in_dhw_vec.end(), out_dims.begin(),
+            [&](int64_t i) -> int { return static_cast<int>(i) * scale; });
+      }
+      auto out_size = ctx.Input<Tensor>("OutSize");
+      if (out_size != nullptr) {
+        auto out_size_data = get_new_data_from_tensor<int>(out_size);
+        if (out_size_data.size() == out_dims.size()) {
+          out_dims = out_size_data;
+        }
+      }
+    }
+    out_dims.insert(out_dims.begin(), in_dims[0]);
+    if (is_channel_last) {
+      out_dims.push_back(in_dims[in_dims.size() - 1]);
+    } else {
+      out_dims.insert(out_dims.begin() + 1, in_dims[1]);
+    }
+    return out_dims;
+  }
+
+ public:
+  void Compute(const framework::ExecutionContext& ctx) const override {
+    const auto& dev_ctx =
+        ctx.template device_context<paddle::platform::MKLDNNDeviceContext>();
+    const auto& mkldnn_engine = dev_ctx.GetEngine();
+
+    const auto* x = ctx.Input<Tensor>("X");
+    std::vector<float> scale_prior;
+    auto* z = ctx.Output<Tensor>("Out");
+
+    auto interp_method = ctx.Attr<std::string>("interp_method");
+    dnnl::algorithm algo = (interp_method == "nearest")
+                               ? dnnl::algorithm::resampling_nearest
+                               : dnnl::algorithm::resampling_linear;
+
+    auto out_dims_vec = ComputeOutputShape(ctx);
+    framework::DDim dim_out = framework::make_ddim(out_dims_vec);
+    z->mutable_data<T>(dim_out, ctx.GetPlace());
+
+    InterpolateMKLDNNHandler<T> handler(algo, dev_ctx, mkldnn_engine,
+                                        ctx.GetPlace(), x, z,
+                                        ctx.OutputName("Out"));
+
+    auto resampling_pd = handler.AcquireForwardPrimitiveDescriptor();
```

Comment:
This is not needed here and is set (fwd_pd_) when MKLDNNHandlerT::isCached() is called.